api/controller.go

```diff
@@ -105,16 +107,22 @@ func (c *Controller) Deploy(serviceName string, d service.Deploy) (*service.Depl
 				controllerLogger.Errorf("Could not create service %v", serviceName)
 				return nil, err
 			}
-		}
-		err = c.createServiceInDynamo(s, d)
-		if err != nil {
-			controllerLogger.Errorf("Could not create service %v in dynamodb", serviceName)
-			return nil, err
+		} else {
+			return nil, errors.New("ECS Service not found and resource creation is disabled")
 		}
 	} else if err != nil {
 		return nil, errors.New("Error during checking whether service exists")
 	} else {
-		c.updateDeployment(d, ddLast, serviceName, taskDefArn, iamRoleArn)
+		serviceExistsInDynamo, err := s.ServiceExistsInDynamo()
+		if err == nil && !serviceExistsInDynamo {
+			err = c.createServiceInDynamo(s, d)
```

Comment:
Does this mean the service will only be created in dynamo on update rather than on initial creation of the service? Should this be outside of the if/else ?

Reply:
The if only refers to whether the ecs service exists or not.
If the ECS Service does exists, then we are in the `else` and we still need to check if the service exists in dynamo, create it if it's not the case.
But indeed I think there is a mistake here where we actually do need to run `c.updateDeployment` even in the case where we just created the service in dynamo